### PR TITLE
Feat/prettier integration

### LINKED ISSUE #129: feat: Add Prettier for Uniform Code Formatting
## Description
The current codebase lacks a consistent code formatting style, which can lead to readability issues and make collaboration more challenging. To address this, we should integrate Prettier, an opinionated code formatter, into the project. Prettier will automatically format the code according to a set of predefined rules, ensuring uniformity across the entire codebase.

## Proposed Solution
Integrate Prettier into the project and configure it to enforce consistent code formatting. This will involve installing Prettier, setting up configuration files, and updating the project's scripts to include Prettier checks.

## Detailed Steps

1. **Install Prettier:**
   - Add Prettier as a development dependency.
     ```bash
     npm install --save-dev prettier
     ```

2. **Create Prettier Configuration File:**
   - Create a `.prettierrc` file in the root of the project to define the formatting rules.
   
   example:
   
     ```json
     {
       "semi": true,
       "singleQuote": true,
       "trailingComma": "es5",
       "printWidth": 80,
       "tabWidth": 2
     }
     ```

3. **Update Package Scripts:**
   - Add scripts to `package.json` to format the code and check for formatting issues.
   
     ```json
     "scripts": {
       "format": "prettier --write \"src/**/*.{js,jsx,ts,tsx,json,css,scss,md}\"",
       "format:check": "prettier --check \"src/**/*.{js,jsx,ts,tsx,json,css,scss,md}\""
     }
     ```

4. **Format Existing Code:**
   - Run the Prettier format script to format the existing codebase.
     ```bash
     npm run format
     ```

5. **Documentation:**
   - Update the project documentation to include instructions on how to use Prettier and the new scripts.

## Benefits
- **Consistent Code Style:** Ensures that all code follows a uniform style, improving readability and maintainability.
- **Automated Formatting:** Reduces the need for manual code formatting, saving time and reducing errors.
- **Improved Collaboration:** Makes it easier for multiple developers to work on the same codebase without style conflicts.

## Additional Notes
- **Editor Integration:** Encourage team members to install Prettier plugins for their code editors to automatically format code on save.
- **Customization:** The Prettier configuration can be customized to fit the team's preferred coding style.

By integrating Prettier into the project, we can enforce a consistent code style across the entire codebase, making it easier to read, maintain, and collaborate on.

### PR DESCRIPTION
**Pull request by shagunsharma6677**
**Description**

This PR fixes #129: Integrated Prettier for consistent code formatting and updated project documentation.

Changes Made:

- Installed Prettier (npm install --save-dev prettier).
- Configured .prettierrc for uniform formatting rules.
- Added formatting (npm run format) and checking (npm run format:check) scripts for JavaScript, TypeScript, JSON, CSS, SCSS, and Markdown files in src.
- Updated documentation with Prettier usage instructions.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -32,12 +32,67 @@ These are, in general, rules that you should be following while contributing to
 ### Pre-requisites
 
 1. Ensure you have the latest version of Pieces OS installed on your system:
+
```
Comment (Arindam200) [inline CONTRIBUTING.md:35]: Remove the Extra line here

**Review by Arindam200 — COMMENTED** [review]
Remove the extra spaces, rest looks good to me

File: CONTRIBUTING.md (hunk 1 above)
Comment (shagunsharma6677) [reply CONTRIBUTING.md]: I have removed extra like.

**Review by shivay-at-pieces — APPROVED** [review]

**Comment by shivay-at-pieces** [comment]
Looks good! Thanks for the contributions!

**Pull request merged**